src/components/DocExplorer.js

```diff
@@ -181,25 +177,33 @@ export class DocExplorer extends React.Component {
     this.showDoc(typeOrField);
   }
 
-  handleSearch = event => {
+  handleSearch = value => {
     this.showSearch({
       name: 'Search Results',
-      searchValue: event.target.value
+      searchValue: value
     });
   }
 }
 
 class SearchBox extends React.Component {
-
   static propTypes = {
     isShown: PropTypes.bool,
-    searchValue: PropTypes.string,
     onSearch: PropTypes.func,
   }
 
-  shouldComponentUpdate(nextProps) {
+  constructor(props) {
+    super(props);
+
+    this.state = { value: null };
+
+    this._debouncedOnSearch = debounce(300, () => {
```

Comment:
300 may be a little slow, have you tested some other values to balance this? My intuition was that it would be closer to 100